### PR TITLE
[IMP] l10n_it, l10n_it_edi: Cleanup fields

### PR DESCRIPTION
**Pull request by lordkrandel**
- `account_tax.l10n_it_vat_due_date`
This field has been superceded by computation on the tax's details. There is no need for an extra explicit field, it just confuses the user, so we're removing it.

- `l10n_it_has_exoneration`
  This field is useless since it can be easily computed: a 0% tax that has a `l10n_it_kind_exoneration` has an exoneration.
  It's not even necessary to leave it as a computed, we just check the presence of the kind of exoneration when necessary.

- `l10n_it_kind_exoneration`
  Renamed to `l10n_it_exempt_reason` following the   `l10n_es_exempt_reason` example

- `l10n_it_exempt_reason`,  `l10n_it_law_reference`
  Moving tax fields from `l10n_it_edi` to `l10n_it`. 
  These are tax details that also apply to Italian Taxes even if they are not sent through the EDI.
  Exempt reason and law reference are set on probably invalid ones just for default.
  They will be changed to valid ones, but we need to add more taxes.

Upgrade PR: odoo/upgrade#5305

Task link: https://www.odoo.com/web#model=project.task&id=3551241
task-3551241

### CODE REVIEW

File: addons/l10n_it/models/account_tax.py [hunk 1]
```diff
@@ -45,22 +35,21 @@ class AccountTax(models.Model):
         ],
         string="Exoneration",
         help="Exoneration type",
-        default="N1",
+        default=False,
```
Comment (jco-odoo) [inline addons/l10n_it/models/account_tax.py:38]:
```suggestion
```

File: addons/l10n_it/models/account_tax.py [hunk 2]
```diff
@@ -45,22 +35,21 @@ class AccountTax(models.Model):
...
                     'refund_repartition_line_ids')
     def _l10n_it_edi_check_exoneration_with_no_tax(self):
         for tax in self:
-            if tax.l10n_it_has_exoneration:
-                if not tax.l10n_it_kind_exoneration or not tax.l10n_it_law_reference or tax.amount != 0:
-                    raise ValidationError(_("If the tax has exoneration, you must enter a kind of exoneration, a law reference and the amount of the tax must be 0.0."))
-                if tax.l10n_it_kind_exoneration == 'N6' and tax._l10n_it_is_split_payment():
+            if tax.country_id.code == 'IT':
+                if (tax.amount_type == 'percent' and tax.amount == 0) and not (tax.l10n_it_exempt_reason and tax.l10n_it_law_reference):
+                    raise ValidationError(_("If the tax amount is 0%, you must the exoneration code and the related law reference."))
```
Comment (jco-odoo) [inline addons/l10n_it/models/account_tax.py:51]: you must what with the exoneration code? 
```suggestion
                    raise ValidationError(_("If the tax amount is 0%, you must enter the exoneration code and the related law reference."))
```

File: addons/l10n_it/models/account_tax.py [hunk 3]
```diff
@@ -45,22 +35,21 @@ class AccountTax(models.Model):
...
                     'invoice_repartition_line_ids',
                     'refund_repartition_line_ids')
     def _l10n_it_edi_check_exoneration_with_no_tax(self):
         for tax in self:
-            if tax.l10n_it_has_exoneration:
-                if not tax.l10n_it_kind_exoneration or not tax.l10n_it_law_reference or tax.amount != 0:
-                    raise ValidationError(_("If the tax has exoneration, you must enter a kind of exoneration, a law reference and the amount of the tax must be 0.0."))
-                if tax.l10n_it_kind_exoneration == 'N6' and tax._l10n_it_is_split_payment():
+            if tax.country_id.code == 'IT':
+                if (tax.amount_type == 'percent' and tax.amount == 0) and not (tax.l10n_it_exempt_reason and tax.l10n_it_law_reference):
```
Comment (jco-odoo) [inline addons/l10n_it/models/account_tax.py:50]:
```suggestion
                if tax.amount_type == 'percent' and tax.amount == 0 and not (tax.l10n_it_exempt_reason and tax.l10n_it_law_reference):
```

File: addons/l10n_it_edi/models/account_chart_template.py [hunk 4]
```diff
@@ -14,8 +14,7 @@ def _post_load_data(self, template_code, company, template_data):
         result = super()._post_load_data(template_code, company, template_data)
         if template_code == 'it':
             self.ref('00eu').write({
-                'l10n_it_has_exoneration': True,
-                'l10n_it_kind_exoneration': 'N3.2',
+                'l10n_it_exempt_reason': 'N3.2',
```
Comment (jco-odoo) [inline addons/l10n_it_edi/models/account_chart_template.py:17]: We do not have an N3.2 in the CoA already now?

**Comment by jco-odoo** [comment]
@lordkrandel Could you squash?

**Comment by lordkrandel** [comment]
Done.

**Comment by jco-odoo** [comment]
@robodoo r+

**Comment by jco-odoo** [comment]
@robodoo r+

**Comment by jco-odoo** [comment]
@robodoo r-

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**